### PR TITLE
Support the compilation of Scala 2.13 StringContext

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

**Comment by nicolasstucki** [comment]
This partially fixes #9438

File: compiler/src/dotty/tools/dotc/transform/PostTyper.scala [hunk 1]
```diff
@@ -148,7 +148,12 @@ class PostTyper extends MacroTransform with IdentityDenotTransformer { thisPhase
           checkInferredWellFormed(tree.tpt)
           val sym = tree.symbol
           if sym.isScala2Macro && !ctx.settings.XignoreScala2Macros.value then
-            if !sym.owner.unforcedDecls.exists(p => !p.isScala2Macro && p.name == sym.name && p.signature == sym.signature) then
+            if !sym.owner.unforcedDecls.exists(p => !p.isScala2Macro && p.name == sym.name && p.signature == sym.signature)
+               // Allow scala.reflect.materializeClassTag to be able to compile scala/reflect/package.scala
+               // At this point we cannot force the symbol of scala.reflect.package$, hence we check it by name
+               // This should be removed on Scala 3.1
+               && sym.showFullName != "scala.reflect.materializeClassTag"
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/transform/PostTyper.scala:155]: We shouldn't rely on the pretty-printer for this. You can use `fullName.toString` instead, but I don't understand why you can't force the symbol, you shouldn't get cycles in PostTyper.

File: compiler/src/dotty/tools/dotc/typer/Typer.scala [hunk 2]
```diff
@@ -3222,21 +3222,23 @@ class Typer extends Namer
         if ((inlined ne tree) && errorCount == ctx.reporter.errorCount) readaptSimplified(inlined)
         else inlined
       }
+      else if tree.symbol.name == nme.f && tree.symbol == defn.StringContext_f then
+        // To avoid forcing StringContext_f when starting compiling StingContex
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/typer/Typer.scala:3226]: typo: "StingContex", also I don't really know what "starting compiling" means.

**Comment by smarter** [comment]
We should update the standard library in our community build to the latest version from upstream.

File: compiler/src/dotty/tools/dotc/typer/Typer.scala (hunk 2 above)
Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/typer/Typer.scala:3226]: should be just "compiling"

File: compiler/src/dotty/tools/dotc/transform/PostTyper.scala (hunk 1 above)
Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/PostTyper.scala]: Removed from this PR

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/PostTyper.scala]: Added it again

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/transform/PostTyper.scala]: But now I check the symbol

**Pull request merged**